src/editor.rs

```diff
@@ -395,41 +392,45 @@ impl Editor {
             self.update_row(self.cursor.y, false);
             // The number of rows has changed. The left padding may need to be updated.
             self.update_screen_cols();
-            self.dirty = true;
-            self.cursor.y -= 1;
-        } else if self.cursor.y == self.rows.len() { self.move_cursor(&AKey::Left); }
-            // If the cursor is located after the last row, pressing backspace is equivalent to
-            // pressing the left arrow key.
+            (self.dirty, self.cursor.y) = (self.dirty, self.cursor.y - 1);
+        } else if self.cursor.y == self.rows.len() {
+            self.move_cursor(&AKey::Left);
+        }
+        // If the cursor is located after the last row, pressing backspace is equivalent to
+        // pressing the left arrow key.
     }
 
     fn delete_current_row(&mut self) {
         if self.cursor.y < self.rows.len() {
             self.rows[self.cursor.y].chars.clear();
             self.update_row(self.cursor.y, false);
-            self.cursor.move_to_next_line();
-            self.delete_char();
+            (_, _) = (self.cursor.move_to_next_line(), self.delete_char());
         }
     }
 
     fn duplicate_current_row(&mut self) {
-        self.copy_current_row(true);
-        self.paste_current_row();
+        (_, _) = (self.copy_current_row(true), self.paste_current_row());
     }
 
     fn copy_current_row(&mut self, is_preserving: bool) {
-        if let Some(row) = self.current_row() { self.copied_row = row.chars.clone(); }
-        if !is_preserving { self.delete_current_row(); }
+        if let Some(row) = self.current_row() {
+            self.copied_row = row.chars.clone();
+        }
+
+        if !is_preserving {
+            self.delete_current_row();
+        }
     }
 
     fn paste_current_row(&mut self) {
-        if self.copied_row.is_empty() { return; }
+        if self.copied_row.is_empty() {
+            return;
+        }
         self.n_bytes += self.copied_row.len() as u64;
         self.rows.insert(self.cursor.y + 1, Row::new(self.copied_row.clone()));
         self.update_row(self.cursor.y + 1, false);
-        self.cursor.y += 1;
-        self.dirty = true;
         // The line number has changed
-        self.update_screen_cols();
+        (self.cursor.y, self.dirty, _) = (self.cursor.y + 1, true, self.update_screen_cols());
```

Comment:
Use only for the assignments, as the rest is unstable.

Reply:
Got it